### DISCUSSION TITLE
How should I import a Tycho "surefire with maven-layout" project in Eclipse ?

### DISCUSSION THREAD

**Discussion by VolatileLongPseudoRandom** [eclipse-tycho/tycho#3503] [Q&A] [answered]
Hello,

I am trying to import a project using the maven layout (src/main/java, no build.properties) which more or less follow this demo project found on the demo folder.

I am using File > Import > Import as Maven plugin with the following path:

- [demo\testing\surefire\with-maven-layout](https://github.com/eclipse-tycho/tycho/tree/main/demo/testing/surefire/with-maven-layout)
- [demo\testing\surefire\with-source-folder](https://github.com/eclipse-tycho/tycho/tree/main/demo/testing/surefire/with-source-folder)

The import of the `with-maven-layout` goes well but the result is fairly wrong:

![image](https://github.com/eclipse-tycho/tycho/assets/156710970/7b44cade-0dd5-41b5-9b70-369afdb44290)

The project JRE should have been Java 17 but Java 5 was selected (!) leading to errors.

There are also some problems shown below, one with the build.properties which I assumed Tycho (?) would have somehow generated it ?

| Description | Resource | Path | Location | Type |
|-------------|----------|------|----------|------|
|build.properties does not exist|surefire-with-maven-layout||Unknown|Plug-in Problem|
|Build path specifies execution environment J2SE-1.5. There are no JREs installed in the workspace that are strictly compatible with this environment.|surefire-with-maven-layout||Build path|JRE System Library Problem|
|Project 'surefire-with-maven-layout' has no explicit encoding set|surefire-with-maven-layout||/surefire-with-maven-layout|No explicit project encoding|
|The compiler compliance specified is 1.5 but a JRE 21 is used|surefire-with-maven-layout||Compiler Compliance|JRE Compiler Compliance Problem|

I know that Tycho is a maven plugin and not responsible for integrating the pom to Eclipse but I don't know which is doing what (m2e-core?) 

For information, I am using an Eclipse 2023-12 (4.30.0 ~ 20231201-2043) and I've also tried tried Eclipse 2024-03

**Comment by laeubi** [comment]
In general m2e is responsible for importing either bare (no `pom.xml`) or classical maven project (with `pom.xml`), so the issue here seems that in the first case m2e is not able to detect the plugin nature of the project and that's why your are seeing issues.

What leads to the question why you need/want to omit `build.properties`, Tycho supports that for cases where one has a more maven oriented project or single modules that follow maven paradigm  but in general the file is quite important for the build as you can configure some more aspects with it that you otherwise have to configure withing you (parent) `pom.xml` and it usually can't be recovered from that (or at least m2e currently does not support it!). 

So it would be good to understand what is the use case of omitting the build.properties in the first place, and if it would be good to enhance m2e in this regards.

**Reply by VolatileLongPseudoRandom** [reply]
> What leads to the question why you need/want to omit build.properties, Tycho supports that for cases where one has a more maven oriented project or single modules that follow maven paradigm but in general the file is quite important for the build as you can configure some more aspects with it that you otherwise have to configure within you (parent) pom.xml and it usually can't be recovered from that (or at least m2e currently does not support it!).

I have honestly no specific reasons to not have a build.properties, although I might say the [documentation here](https://tycho.eclipseprojects.io/doc/latest/TestingBundles.html#tycho-surefire-plugin) and the example both lead me into thinking we could have Eclipse plugin following maven convention without such file. 

I don't know if the demo project are meant to be real use case or integration project for Tycho, but if they are the former, I dare to say they should import as maven project without any hitches  in Eclipse m2e-core - especially if you want to develop Eclipse plugin :)

And the case the example with maven layout should be valid, perhaps I should try to create an issue at m2e-core.

**Reply by laeubi** [reply, marked as answer]
Yes it is possible but if you not only want to build it (like in the demo) for real world development you most probably need to configure much more in the `pom.xml` ... so there are some project that go maven first, but you will get more convenient and less configuration intensive results if you just start from the IDE and then just use that.

Tycho can literally everything then derive from that so your build can have a single almost empty pom then:
- https://tycho.eclipseprojects.io/doc/latest/StructuredBuild.html

The mentions layout here
- https://tycho.eclipseprojects.io/doc/latest/TestingBundles.html

more refers to the `src/main` and `src/test` using as source folders than to omit all metadatafiles (like `build.properties`), because in such case you still need to configure the folder in `build.properties` to make PDE pick them up in the IDE.

**Reply by VolatileLongPseudoRandom** [reply]
Ok. That's mostly answer my question :) thanks for the replies
